### PR TITLE
HS-775 - Add a configurable deadline to all gRPC call

### PR DESCRIPTION
**Pull request by BenjaminJ**
default value 60000 milliseconds

HS-775

## Description
Add a configurable deadline to all gRPC call, it's default value is 60 seconds. see [https://grpc.io/blog/deadlines/](url)

## Jira link(s)
- https://issues.opennms.org/browse/HS-775

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcClient.java [hunk 1]
```diff
@@ -91,6 +93,6 @@ public void onCompleted() {
         } catch (Exception e) {
             throw new RuntimeException("Failed to call minion", e);
```
Comment (yangli2014) [inline inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcClient.java:94]: It it better to use future.completeExceptionally()

File: inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcClient.java [hunk 2]
```diff
@@ -91,6 +93,6 @@ public void onCompleted() {
...
         }
-        return future.orTimeout(60, TimeUnit.SECONDS);
+        return future.orTimeout(deadline, TimeUnit.MILLISECONDS);
```
Comment (yangli2014) [inline inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcClient.java:96]: Do we still need timeOut here after we added the deadline?

**Review by cgorantla — COMMENTED** [review]
As discussed in the call today, we can remove changes on Minion but keep other changes.

File: inventory/src/test/java/org/opennms/horizon/inventory/component/MinionRpcClientTest.java [hunk 3]
```diff
@@ -92,7 +92,7 @@ public void request(RpcRequestProto request, StreamObserver<RpcResponseProto> re
             .addService(mockRequestService)
             .directExecutor().build().start());
         ManagedChannel channel = grpcCleanup.register(InProcessChannelBuilder.forName(MinionRpcClientTest.class.getName()).directExecutor().build());
-        client = new MinionRpcClient(channel, (ctx) -> Optional.ofNullable(GrpcConstants.TENANT_ID_CONTEXT_KEY.get()));
+        client = new MinionRpcClient(channel, (ctx) -> Optional.ofNullable(GrpcConstants.TENANT_ID_CONTEXT_KEY.get()), 1000);
```
Comment (cgorantla) [inline inventory/src/test/java/org/opennms/horizon/inventory/component/MinionRpcClientTest.java:95]: deadline `1000` is too tight. May be `5000` will be better ?

**Review by cgorantla — COMMENTED** [review]
LGTM, minor comments!

File: inventory/src/test/java/org/opennms/horizon/inventory/component/MinionRpcClientTest.java (hunk 3 above)
Comment (BenjaminJ) [reply inventory/src/test/java/org/opennms/horizon/inventory/component/MinionRpcClientTest.java:95]: Did you mean 500 ?

**Pull request merged**